### PR TITLE
Executable samples to export

### PR DESCRIPTION
**Pull request by parodi-smeup**
## Description

Include a summary of the change.

Related to # (issue)

## Checklist:
- [ ] There are tests regarding this feature
- [ ] The code follows the Kotlin conventions (run `./gradlew ktlintCheck`)
- [ ] The code passes all tests (run `./gradlew check`)
- [ ] There is a specific documentation in the `docs` directory

### CODE REVIEW

**Review by lanarimarco — CHANGES_REQUESTED** [review]

File: examples/src/main/kotlin/com/samples/jariko/RpgSamples.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import com.smeup.rpgparser.execution.Configuration
+import com.smeup.rpgparser.execution.JarikoCallback
+import com.smeup.rpgparser.execution.executePgmWithStringArgs
+import com.smeup.rpgparser.execution.getProgram
+import com.smeup.rpgparser.interpreter.ISymbolTable
+import com.smeup.rpgparser.jvminterop.JavaSystemInterface
+import com.smeup.rpgparser.rpginterop.DirRpgProgramFinder
+import java.io.File
+
+/**
```
[the remark this answers is not in the document]
Comment (parodi-smeup) [reply examples/src/main/kotlin/com/samples/jariko/RpgSamples.kt:12]: done

File: examples/src/main/kotlin/com/samples/jariko/RpgSamples.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ * @param programArgs parm to pass to rpgle
+ * @param jarikoCallback instance of JarikoCallback
+ *
+ * or:
+ *
+ * @param programSource source of rpgle program
+ * @param programArgs parm to pass to rpgle
+ * @param jarikoCallback instance of JarikoCallback
+ */
+
```
[the remark this answers is not in the document]
Comment (parodi-smeup) [reply examples/src/main/kotlin/com/samples/jariko/RpgSamples.kt]: done

File: examples/src/main/kotlin/com/samples/jariko/RpgSamples.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    val programArgs = listOf(fibonacciOf.toString())
+    val jarikoCallback = JarikoCallback(
+            exitInRT = { false },
+            onExitPgm = { _: String, symbolTable: ISymbolTable, _: Throwable? ->
+                println(symbolTable["FINAL_VAL"].asInt().value)
+            }
+    )
+    execWithCallback(programPath, programName, programArgs, jarikoCallback)
+}
+
```
[the remark this answers is not in the document]
Comment (parodi-smeup) [reply examples/src/main/kotlin/com/samples/jariko/RpgSamples.kt:29]: done

File: examples/src/main/kotlin/com/samples/jariko/RpgSamples.kt [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    )
+    val systemInterface = JavaSystemInterface()
+    val commandLineProgram = getProgram(programSource, systemInterface)
+    commandLineProgram.singleCall(programArgs, configuration = configuration)
+    println("... done.")
+}
+
+fun main(args: Array<String>) {
+    // Sample of fibonacciOf(15) using local rpgle source file "fibonacci.rpgle"
+    fibonacciOf(15)
```
[the remark this answers is not in the document]
Comment (parodi-smeup) [reply examples/src/main/kotlin/com/samples/jariko/RpgSamples.kt:77]: not a "main" anymore, changed to a generic callable fun with non parameres

File: examples/src/main/META-INF/MANIFEST.MF [hunk 5]
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
```
[the remark this answers is not in the document]
Comment (parodi-smeup) [reply examples/src/main/META-INF/MANIFEST.MF]: removed

File: examples/build.gradle [hunk 6]
```diff
@@ -24,3 +24,14 @@ dependencies {
   testCompile "org.jetbrains.kotlin:kotlin-test-junit:$kotlinVersion".toString()
   testCompile 'junit:junit:4.12'
 }
+
+//Creates smeup-rpg/rpgJavaInterpreter-core/build/libs/rpgJavaInterpreter-core-runnable-samples-all.jar
+task fatSamplesJar(type: Jar) {
+    manifest {
```
[the remark this answers is not in the document]
Comment (parodi-smeup) [reply examples/build.gradle]: removed

File: examples/src/main/kotlin/com/samples/jariko/RpgSamples.kt [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+ * @param fibonacciOf integer to calculate fibonacci series
+ */
+fun fibonacciOf(fibonacciOf: Int) {
+    val programPath = "examples/src/main/kotlin/com/samples/jariko"
+    val programName = "fibonacci.rpgle"
+    val programArgs = listOf(fibonacciOf.toString())
+    val jarikoCallback = JarikoCallback(
+            exitInRT = { false },
+            onExitPgm = { _: String, symbolTable: ISymbolTable, _: Throwable? ->
+                println(symbolTable["FINAL_VAL"].asInt().value)
```
Comment (lanarimarco) [inline examples/src/main/kotlin/com/samples/jariko/RpgSamples.kt:24]: Printing of the calculated number is not what I would. I would like the function to return the calculated number

**Review by lanarimarco — CHANGES_REQUESTED** [review]
There is a little change request in function fibonacciOf that should return the calculated number.

**Review by lanarimarco — APPROVED** [review]
Ok

**Pull request merged**